### PR TITLE
CAT-1066: Add math operation for exponentiations(power)

### PR DESCRIPTION
**Pull request by bernadettespieler**

### CODE REVIEW

File: catroid/src/org/catrobat/catroid/formulaeditor/FormulaElement.java [hunk 1]
```diff
@@ -354,6 +354,12 @@ private Object interpretFunction(Functions function, Sprite sprite) {
 				return doubleValueOfLeftChild == null ? 0d : java.lang.Math.toDegrees(Math.atan(doubleValueOfLeftChild));
 			case EXP:
 				return doubleValueOfLeftChild == null ? 0d : java.lang.Math.exp(doubleValueOfLeftChild);
+			case POWER:
+				try {
+					return java.lang.Math.pow(doubleValueOfLeftChild, doubleValueOfRightChild);
+				} catch (Exception e) {
+					return 0d;
+				}
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/formulaeditor/FormulaElement.java]: To be consistent with all other cases, use `(doubleValueOfLeftChild == null || doubleValueOfRightChild == null) ? 0d :..` as well.

If there aren't unit tests, please create some.

**Comment by thmq** [comment]
ok to test

File: catroid/src/org/catrobat/catroid/formulaeditor/InternFormulaKeyboardAdapter.java [hunk 2]
```diff
@@ -104,6 +104,9 @@
 				return buildSingleParameterFunction(Functions.ARCTAN, InternTokenType.NUMBER, "0");
 			case R.string.formula_editor_function_exp:
 				return buildSingleParameterFunction(Functions.EXP, InternTokenType.NUMBER, "1");
+			case R.string.formula_editor_function_power:
+				return buildDoubleParameterFunction(Functions.POWER, InternTokenType.NUMBER, "2", InternTokenType
+						.NUMBER, "3");
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/formulaeditor/InternFormulaKeyboardAdapter.java]: Please break the line at a better position :smile_cat:

File: catroid/src/org/catrobat/catroid/ProjectManager.java [hunk 3]
```diff
@@ -444,6 +444,7 @@ public int getCurrentSpritePosition() {
 		return project.getSpriteList().indexOf(currentSprite);
 	}
 
+	//Function apperantly never used please check if necessary!
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/ProjectManager.java]: The [code inspection for 'Unused declaration'](http://stackoverflow.com/a/6587932/2774399) will tell you another 241 unused items. :wink:
Please remove this comment. Feel free to create a Jira ticket to remove unused code. Beware, some code is used by reflection but may be reported as well.
I also would :heart: to see the IntelliJ Code Inspection tool running on Jenkins [CAT-1672](https://jira.catrob.at/browse/CAT-1672)

**Pull request closed without merging**